### PR TITLE
Finalized figs

### PR DESCRIPTION
**Pull request by axiomcura**
This pull request introduces Figure 2 to the Pycytominer paper, giving a view of the workflow, model performance, and its application to JUMP. 

**Additional Notes**
Threre are some bug fixes encountered during the notebook execution and the integration of extra functions to help create figures and tables.

### CODE REVIEW

**Comment by gwaybio** [comment]
Overall, this is a much improved figure. Kudos! However, there is still quite a bit of work needed before it is publication ready. Each individual element below is a nitpick, but added up together, they will make a big difference in improving our story.

## Figure nitpicks

- White space management. 
  - There is a bunch of space between the tag label `A` and the figure. 
  - There is also quite a bit of space between panel A and the second row of figures
  - Panel C is also squeezed to make a lot more space between the tag label `C` and the figure
  - The bottom of the figure only contains the x axis label in panel `C`
-  Panel A
   - The "Evaluate model" section text is super small. Perhaps you can drop the confusion matrix icon and increase the PR curve text to the same size as Train/Test in the "Data splitting" section.
   - The arrow from "JUMP dataset" to "Trained model" should probably be reversed, right? (pointing away from trained model and toward JUMP?)
   - Is there a way to efficiently note that pycytominer also processed the JUMP features?
   - Do you think it is necessary to specify the Cell injury dataset somewhere here?
- Panel B
  - The font sizes for axis titles, axis tick labels, legend titles, and legend text are still a bit small (facet labels are probably as large as they can be)
  - I like the F1 score box, but in many cases, it overshadows the lines, which is not good. Please move each box within each facet to a custom location to avoid overlapping the lines.
  - There remains some title case labels here (specifically, "Data Split" and "Not Shuffled"). please fix
- Panel C
  - The text labels in the squares are very hard to read. See "figure organization" section below, which I think will solve this problem
  - I think the font size everywhere else looks great now, but you will have to double check this after you fix figure organization
  - There remains some title case labels here (specifically, "True Class", "Predicted Class", "Plate Holdout", and "Well Holdout"). please fix
  - The ordering of the facets are a bit wonky. It should be "Train", "Test", "Well holdout", "Plate holdout" -> in order of expected decreasing performance.
- Panel D
  -  There remains some title case labels here (specifically, "Cytoskeletal Injury Probability", "Predicted Injuries", "Model Type", and "Not Shuffled"). please fix
  - Please decrease the alpha channel (transparency) so we can see the blue line behind the red line in the first panel
  - The x axis is illegible. See "figure organization" section below, which I think will solve this problem

## Figure organization

Here's what I think will finalize the figure:

- Move panel D to below panel B
- Add blank plot to below panel C

Let's chat about all of this if you have any questions

File: notebooks/3.jump-analysis/3.jump_analysis.py [hunk 1]
```diff
@@ -0,0 +1,453 @@
...
+    split_meta_and_features,
+)
+
+# ## Setting up parameters and paths
+
+# In[2]:
+
+
+# setting up paths and output paths
+results_dir = pathlib.Path("../../results")
```
Comment (MikeLippincott) [inline notebooks/3.jump-analysis/3.jump_analysis.py]:
```suggestion
results_dir = pathlib.Path("../../results").resolve()
```

File: notebooks/3.jump-analysis/3.jump_analysis.py [hunk 2]
```diff
@@ -0,0 +1,453 @@
...
+]
+well_counts_df
+
+
+# In[9]:
+
+
+# this is our ground truth since this compound is also found in the cell-injury dataset
+jump_cyto_injury_df = shared_jump_df.loc[
+    shared_jump_df["Metadata_InChIKey"] == "IAKHMKGGTNLKSZ-INIZCTEOSA-N"
```
Comment (MikeLippincott) [inline notebooks/3.jump-analysis/3.jump_analysis.py]: what is it??

File: notebooks/3.jump-analysis/3.jump_analysis.py [hunk 3]
```diff
@@ -0,0 +1,453 @@
...
+
+
+# In[20]:
+
+
+jump_meta = shared_jump_df[["Metadata_Plate", "Metadata_Well", "Metadata_pert_iname"]]
+jump_meta["pred_injury"] = [
+    injury_decoder[str(injury_code)] for injury_code in y_pred.tolist()
+]
+jump_meta["probability"] = y_proba.max(axis=1).tolist()
```
Comment (MikeLippincott) [inline notebooks/3.jump-analysis/3.jump_analysis.py]: why max here??? Getting the max proba for....?

File: notebooks/3.jump-analysis/3.jump_analysis.py [hunk 4]
```diff
@@ -0,0 +1,453 @@
...
+    compression="gzip",
+    index=False,
+)
+jump_meta
+
+
+# In[22]:
+
+
+jump_meta.loc[jump_meta["Metadata_Plate"] == "BR00117006"][
```
Comment (MikeLippincott) [inline notebooks/3.jump-analysis/3.jump_analysis.py]: Without sounding rude, why is this here after exporting the data? If it is there for a reason consider documenting it

File: notebooks/3.jump-analysis/nbconverted/3.jump_analysis.py [hunk 5]
```diff
@@ -422,3 +413,33 @@
...
+
+
+# In[20]:
+
+
+jump_meta = shared_jump_df[["Metadata_Plate", "Metadata_Well", "Metadata_pert_iname"]]
+jump_meta["pred_injury"] = [
+    injury_decoder[str(injury_code)] for injury_code in y_pred.tolist()
+]
+jump_meta["probability"] = y_proba.max(axis=1).tolist()
```
Comment (MikeLippincott) [inline notebooks/3.jump-analysis/nbconverted/3.jump_analysis.py:438]: same here why max??

File: notebooks/4.visualization/Figure_2_panel.r [hunk 6]
```diff
@@ -0,0 +1,582 @@
...
+suppressPackageStartupMessages(suppressWarnings(library(lintr))) # linting
+suppressPackageStartupMessages(suppressWarnings(library(ggplot2))) # plotting
+suppressPackageStartupMessages(suppressWarnings(library(dplyr))) # data manipulation
+suppressPackageStartupMessages(suppressWarnings(library(reshape2))) # data manipulation
+suppressPackageStartupMessages(suppressWarnings(library(ggridges))) # ridgeline plots
+suppressPackageStartupMessages(suppressWarnings(library(RColorBrewer))) # color palettes
+suppressPackageStartupMessages(suppressWarnings(library(patchwork))) # color palettes
+suppressPackageStartupMessages(suppressWarnings(library(stringr))) # color palettes
+suppressPackageStartupMessages(library(grid))
+suppressPackageStartupMessages(library(png))
```
Comment (MikeLippincott) [inline notebooks/4.visualization/Figure_2_panel.r:12]: consider adding comments here

File: notebooks/4.visualization/Figure_2_panel.r [hunk 7]
```diff
@@ -0,0 +1,582 @@
...
+f1_score_path <- file.path("../../results/2.modeling/all_f1_scores.csv.gz")
+
+# precision and recall curve path
+pr_file_path <- file.path("../../results/2.modeling/precision_recall_scores.csv.gz")
+
+# probability plot
+cyto_proba_path <- file.path("../../results/3.jump_analysis/cytoskeletal_proba_scores.csv.gz")
+
+# injury probabilities
+injury_proba_path = file.path("../../results/3.jump_analysis/all_injury_proba.csv.gz")
```
Comment (MikeLippincott) [inline notebooks/4.visualization/Figure_2_panel.r]:
```suggestion
injury_proba_path <- file.path("../../results/3.jump_analysis/all_injury_proba.csv.gz")
```

consistency

File: notebooks/4.visualization/Figure_2_panel.r
Comment (MikeLippincott) [inline notebooks/4.visualization/Figure_2_panel.r:1]: this file should be in the nbconverted path. Ensure that the scripts from notebooks are all converted and in the correct paths

File: notebooks/4.visualization/figures/Final_Figure2.png
Comment (MikeLippincott) [inline notebooks/4.visualization/figures/Final_Figure2.png:1]: Looks great! 

Nit picks:
D: x-axis does not go to 1
Lots of white space in bottom right

File: notebooks/4.visualization/figures/supplemental/sfig4_all_injury_probabilities.png
Comment (MikeLippincott) [inline notebooks/4.visualization/figures/supplemental/sfig4_all_injury_probabilities.png:1]: Looks good. The titles are overlapping though. The visualize and save widths and heights were probably different.

**Review by MikeLippincott — COMMENTED** [review]
LGTM, I made comments. I think this one might need an approval from @gwaybio ?

Figures look a lot better though!

**Comment by axiomcura** [comment]
Thanks for the suggestions!

The figures have been approved by Greg already via zoom call. Just needed someone else to view the code.

Thanks again!

File: notebooks/4.visualization/figures/supplemental/sfig4_all_injury_probabilities.png
Comment (axiomcura) [reply notebooks/4.visualization/figures/supplemental/sfig4_all_injury_probabilities.png:1]: hmm could be that I didn't add the correct image dimensions when saving it! Good catch!  Thanks!

File: notebooks/3.jump-analysis/3.jump_analysis.py (hunk 2 above)
Comment (axiomcura) [reply notebooks/3.jump-analysis/3.jump_analysis.py]: Documentation updated! Here's the explanation:

The InChIKey "IAKHMKGGTNLKSZ-INIZCTEOSA-N" represents the compound found in both the cell-injury and JUMP-CP datasets.

Therefore the wells containing this compound (represented by the InChIKey) is then removed from the original JUMP-CP dataset and as a result, we used this wells as our ground truth.


Hopefully this made sense! Thanks!

File: notebooks/3.jump-analysis/3.jump_analysis.py (hunk 4 above)
Comment (axiomcura) [reply notebooks/3.jump-analysis/3.jump_analysis.py]: Not at all sounding! However....Ehhh... This was not supposed be here. I thought I deleted this cell... I might have hit the undo button too many time and brought the cell from the dead. 😨 

Thanks for the catch! Fixed! ✅

File: notebooks/4.visualization/figures/Final_Figure2.png
Comment (axiomcura) [reply notebooks/4.visualization/figures/Final_Figure2.png:1]: Yup!, unfortunately, we couldn't find a way to organize the figures without leaving that blank space while maintaining the quality of the subfigures.

We attempted to fit B, C, and D in the same row, but (D) did not look great. Therefore, we decided to format it this way.

If you have any suggestions for addressing this issue, I am all ears! 😄

File: notebooks/4.visualization/Figure_2_panel.r
Comment (axiomcura) [reply notebooks/4.visualization/Figure_2_panel.r:1]: nbconverted path? You mean the directory! if so, done! ✅

**Comment by axiomcura** [comment]
Merging! Thanks for the review!
